Graphic or logo in the header

der-schmelle2 opened the pull request:
Changes proposed in this pull request:
* Section about how to place a graphic or logo in the page header (or footer)

der-schmelle2 committed:
Section about how to place a graphic or logo in the header of an document

der-schmelle2 committed:
Add myself as author

der-schmelle2 committed:
Add a link within the section

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+   xmlns="http://docbook.org/ns/docbook"
+   xmlns:xi="http://www.w3.org/2001/XInclude"
+   xmlns:xlink="http://www.w3.org/1999/xlink">
+   <title>Inserting Graphics to your Page Headers or Footers</title>
```
Maybe rephrase it to "Inserting Graphics into Page Headers and Footer"? 
Hmn, thinking about if "Logos" instead of "Graphics" would be more appropriate?

der-schmelle2 replied:
Done. 
New title is: "Inserting Logos into Page Headers and Footers"

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+         <keyword>page header</keyword>
+         <keyword>block</keyword>
+         <keyword>format</keyword>
+         <keyword>page</keyword>
```
Add additional keywords like "header" and "footer". I would remove block, it's quite general and as such almost useless.

der-schmelle2 replied:
done.
Changed keywords to:
     <keywordset>
       <keyword>page header</keyword>
       <keyword>page footer</keyword>
       <keyword>header</keyword>
       <keyword>footer</keyword>
       <keyword>format</keyword>
       <keyword>page</keyword>
      </keywordset>

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+
+   <section role="solution">
+      <title>Solution</title>
+      <para>First of all, you need to create a customization layer in your 
```
Basically, this is all nice and correct. However, not everybody read the book from start to end. Usually, users see a topic, jump in and read. That's why you cannot expect they know all the terms and workflows (for example, "customization layer").

I would suggest to structure the solution section like this:

1. Explain first, why you need logos/graphics in the header or footer. What's the benefits?
1. Present the user a very general idea what you will show. In your case, mention that the following stylesheet will create a logo on the right side only. 
1. Optionally, you can show the result as a nice graphic. That way, the reader has at least an idea what it is all about.

After you did your "introduction", give the reader a procedure with a step-by-step instruction how to create this solution. You can use [Creating Permalinks](http://doccookbook.sourceforge.net/html/en/dbc.html.permalinks.html) as a blueprint or the XML file `en/xml/html/topic.permalinks.xml`.

Appyling that style to this topic would mean:

1. Give your changes a file name, for example, `logo-right-header.xsl`. Basically, it can be any name as long as it is  easy to remember and reflects its purpose.
1. Make sure, your stylesheet is syntactically correct and starts with the appropriate root element (`xsl:stylesheet`).
1. Apply steps 1-2 from the "Creating Permalinks" topic to your file.
1. Finish your last step with the phrase: "Use your customization layer to transform your DocBook document into FO."

I guess, it would only need 3 to 4 steps.

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+      <title>Solution</title>
+      <para>First of all, you need to create a customization layer in your 
+         customization stylesheet containing the following code:</para>
+      <programlisting>&lt;!-- Path and name of your graphic file-->
```
Two issues:

* add the attribute `language="xml"` into `<programlisting>`.
* Add the missing XSLT header (root element, namespaces etc.)

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+      <para>First of all, you need to create a customization layer in your 
+         customization stylesheet containing the following code:</para>
+      <programlisting>&lt;!-- Path and name of your graphic file-->
+&lt;xsl:param name="header.image.filename">{$img.src.path}your-graphic.svg&lt;/xsl:param>
```
That won't work because you've included it as text. Text won't be replaced, you need the `select` attribute:

```xml
<xsl:param name="header.image.filename" select="concat($img.src.path, 'your-graphic.svg')"/>
```

Another idea from an implementation perspective: would it better to separate the logo file name from any path? In other words, the parameter `header.image.filename` could just hold the base file name (without any paths). The exact location is created in line 88 with the above `concat(...)` expression. What do you think?

der-schmelle2 replied:
I'm not sure what you mean. Do you mean this:
    <xsl:param name="header.image.filename">your-graphic.svg>/xsl:param>
    <xsl:param name="header.image.filename" select="concat($img.src.path, 'your-graphic.svg')"/>

tomschr replied:
My idea was that `header.image.filename` contains only the file name and nothing else:

```xml
<xsl:param name="header.image.filename">your-graphic.svg</xsl:param>
```

The complete path would be built at the position where it is needed by concatenating it with `$img.src.path`. Does that make any sense? :wink:

der-schmelle2 replied:
Ok. That makes sense. Thanks!
The complete solution looks like this?
    <xsl:param name="img.src.path">./Images/</xsl:param>
    <xsl:param name="header.image.filename">your-graphic.svg</xsl:param>
    xsl:param name="header.image.filename" select="concat($img.src.path, 'your-graphic.svg')"/>
I‘m not sure if this is correct, because 'your-graphic.svg' after concatenation is incorrect.

tomschr replied:
Almost. :wink:  These lines are correct:

```xml
<xsl:param name="img.src.path">./Images/</xsl:param>
<xsl:param name="header.image.filename">your-graphic.svg</xsl:param>
```

A little bit deeper you use the parameter `header.image.filename` in this code:

```xml

&lt;xsl:call-template name="fo-external-image"><co xml:id="co.topic.graphic-pageheader.template"/>
     &lt;xsl:with-param name="filename" select="concat($img.src.path, $header.image.filename)"/><co xml:id="co.topic.graphic-pageheader.file"/>>
&lt;/xsl:call-template>
```
Keep in mind the following changes in comparison to your code:

* `xsl:param` inside `xsl:call-template` has to be `xsl:with-param`.
* the parameter for the template `fo-external-image` has to be `filename`, not `header.image.filename`.
* you pass the complete path in the `select` attribute of `xsl:with-param`.

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+                   &lt;xsl:attribute name="src"><co xml:id="co.topic.graphic-pageheader.attribute"/>
+                        &lt;xsl:call-template name="fo-external-image"><co xml:id="co.topic.graphic-pageheader.template"/>
+                            &lt;xsl:with-param name="filename" select="$header.image.filename"/>
+                            <co xml:id="co.topic.graphic-pageheader.file"/>>
```
Remove the line break and add the `<co/>` right after the end of line 88.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+
+            &lt;xsl:when test="$position='right'"><co xml:id="co.topic.graphic-pageheader.right"/>
+                &lt;!-- Same for odd, even, empty, and blank sequences -->
+                &lt;fo:external-graphic content-height="25mm"><co xml:id="co.topic.graphic-pageheader.height"/>
```
Hmn... hard-coded values are not always as constant as we wish. :wink: I would suggest to outsource it into a parameter. This gives the user the chance to change it as needed.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+               of all pages except title pages.</para>
+         </callout>
+         <callout arearefs="co.topic.graphic-pageheader.variables">
+            <para>The content of the necessary variables needs to be cleared.</para>
```
No, it's the default value. This value is used, when the caller doesn't define the parameter.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+            <para>The content of the necessary variables needs to be cleared.</para>
+         </callout>
+         <callout arearefs="co.topic.graphic-pageheader.block">
+            <para>A <tag>block</tag>-element creates a rectangle area on a page
```
* block -> fo:block

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+         </callout>
+         <callout arearefs="co.topic.graphic-pageheader.block">
+            <para>A <tag>block</tag>-element creates a rectangle area on a page
+            which can contain <tag>lists</tag>, <tag>tables</tag> or <tag>graphics</tag>
```
Avoid these tag elements for list, tables, and graphics. They are not specific elements, but concepts.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+            .</para>
+         </callout>
+         <callout arearefs="co.topic.graphic-pageheader.blank">
+             <para>Here you can place an banner like '<emphasis>This page is
```
* place an banner -> place a banner; or even better use "insert text like..."
* Remove the apostroph before `<emphasis>`

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+         </callout>
+         <callout arearefs="co.topic.graphic-pageheader.blank">
+             <para>Here you can place an banner like '<emphasis>This page is
+                 intentionally left blank'</emphasis> or so.</para>
```
Remove the "or so".

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+          </callout>
+          <callout arearefs="co.topic.graphic-pageheader.right">
+              <para>This is the place where we want to place our graphic. The
+              graphic should be visible on all pages on the right side.</para>
```
This statement should be added *before* you present your stylesheet.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+          </callout>
+          <callout arearefs="co.topic.graphic-pageheader.height">
+              <para>The graphic picture is too big, so we limit his height to 25mm.
+              When you specify only one parameter like <tag>height</tag>or<tag>width</tag>,
```
Add spaces between "height", "or", and "width".

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+              graphic should be visible on all pages on the right side.</para>
+          </callout>
+          <callout arearefs="co.topic.graphic-pageheader.height">
+              <para>The graphic picture is too big, so we limit his height to 25mm.
```
Maybe rephrase the sentence as:

To avoid pictures which are too big for the headline, we limit its height to 25mm.

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+          </callout>
+          <callout arearefs="co.topic.graphic-pageheader.attribute">
+              <para>We will give the graphic which will be used the attribute.
+                  <emphasis>src</emphasis>.</para>
```
That isn't an English sentence. :wink: Maybe something like this:

```
The filename is passed on the <tag class="attribute">src</tag> attribute.
```

der-schmelle2 replied:
You're right. :-)
Thanks!

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+      <title>Discussion</title>
+       <para>If you wish to place your graphic within the footer, replace 
+           <tag>header.content</tag>with<tag>footer.content</tag>.</para>
+   </section>
```
The discussion is a bit... well, *very* short. :wink: 

Usually, I apply the following approach for each topic:

* First, I state the problem
* Second, I present a solution which is easy, general, or short.
* Third, I present further additions. What can the reader do with that solution? Maybe he has further needs, wishes, or problems?

Overall, I try to make the solution more general, but the discussion more specific.

As a reader, I would expect some additional ideas. Some ideas that come to my mind and could be added:

* Why do I need that? What are the reasons? At least, this is very important to give the reader a purpose.
  This could be layout issues, corporate identity, etc.
* What should I do when I have a double sided document?
   Your solution does not take into account double sided documents, right?
* What should I change, if I want my logo appear left on a verso page and right on a recto page?
* Are there any recommendations prior to this stylesheets in regards to file size, file format etc.?
* Could I change the logo depending on the chapter?
  For example, the logo file name could be added in the info element of the respective chapter

Of course, you don't have to describe all of them. It's just an idea what *can* be included.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+                 intentionally left blank'</emphasis> or so.</para>
+         </callout>
+          <callout arearefs="co.topic.graphic-pageheader.center">
+              <para>This is a good place for the chapter title.</para>
```
How do you do that? That can be a good place for the discussion.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,170 @@
+      <title>Solution</title>
+      <para>First of all, you need to create a customization layer in your 
+         customization stylesheet containing the following code:</para>
+      <programlisting>&lt;!-- Path and name of your graphic file-->
```
Further ideas which makes handling of such files easier:

* Use `<example>` around your `programlisting` element.
   This makes it easier to refer to it and you can give it a title and name (see [Creating Permalinks, Example 5.1](http://doccookbook.sourceforge.net/html/en/dbc.html.permalinks.html)
* Outsource your XSLT stylesheet into a separate file and `xi:include` it in your XML source.
  This helps in making example files and (more important!) gives our readers the chance to download this file
* Reduces the size of your XML file.

If you like this approach, I'd recommend the following:

1. Create a directory `logo` under `fo`.
1. Save you XSLT stylesheet in `fo/logo/YOUR_STYLESHEET.xsl`. Replace the placeholder `YOUR_STYLESHEET.xsl` with some useful name.
1. Replace the content of your stylesheet code in your XML file with the following content:

```xml
<example xml:id="ex.fo.XXXXX">
   <title>Some good title (<filename>YOUR_STYLESHEET.xsl</filename>)</title>
   <info>
        <output xmlns="urn:x-toms:docbook-ext">
          <filename>logo/YOUR_STYLESHEET.xsl</filename>
        </output>
   </info>
   <programlistingco>
      <areaspec>
            <!-- Add your callouts through <area/> -->
      </areaspec>
      <programlisting language="xml" linenumbering="numbered"
          ><xi:include href="logo/YOUR_STYLESHEET.xsl" parse="text"
          /></programlisting>
   </programlistingco>
</example>
```

der-schmelle2 replied:
Done, but with a similar way. I've placed all necessary files below the folder "graphic-pageheader/".

tomschr reviewed (changes requested):
Thanks Peter for your contributions! :+1:

Don't worry about the excessive comments. I reviewed it thoroughly and that's the result. Hope you don't feel overwhelmed. :wink: 

If you have questions or don't agree with one of my opinions, let me know and we can discuss this.

der-schmelle2 committed:
Made alterations suggested through the review by @tomschr - Part I

der-schmelle2 committed:
Imagefile handling within programlisting

der-schmelle2 committed:
Build example stylesheet file

der-schmelle2 committed:
Merge branch develop into feature/pageheader-graphic

der-schmelle2 committed:
Merge branch feature/pageheader-graphic into develop

der-schmelle2 committed:
Document redesign to prozedural steps

Modified stylesheet file

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -57,7 +57,8 @@
     &lt;!-- Path and name of your graphic file-->
-    &lt;xsl:param name="header.image.filename">your-graphic.svg&lt;/xsl:param>
+    &lt;xsl:param name="img.src.path">./graphic-pageheader/>
+    &lt;xsl:param name="header.image.filename">Opensource.svg&lt;/xsl:param>
```
This contains a syntax error:

```xml
&lt;xsl:param name="img.src.path">./graphic-pageheader&lt;/xsl:param>
```

der-schmelle2 committed:
Syntax error in programlisting
* Topics for discussion added

@tomschr Please review the "Solution" section, "Discussion" section will follow

der-schmelle2 committed:
Syntax error corrected, second time

http://docbook.sourceforge.net/release/xsl/1.76.1/doc/fo/img.src.path.html
**his prefix is not applied to any filerefs that start with "/" or contain "//:".**
;-)

tomschr commented on en/xml/fo/graphic-pageheader/graphic-pageheader.xsl:
```diff
@@ -0,0 +1,59 @@
+  xmlns:d="http://docbook.org/ns/docbook">
+
+  <!-- Path and name of your graphic file-->
+    <xsl:param name="img.src.path">./graphic-pageheader/</xsl:param>
```
I think we should/need to rename this parameter because there is already the same in the original DocBook stylesheets. Maybe something like `logo.src.path`? By default, it's empty:

```xsl
<xsl:param name="logo.src.path"/>
```

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/graphic-pageheader/graphic-pageheader.xsl:
```diff
@@ -0,0 +1,59 @@
+                    <fo:external-graphic content-height="1in">
+                       <xsl:attribute name="src">
+                            <xsl:call-template name="fo-external-image">
+                                <xsl:with-param name="filename" select="concat($img.src.path, $header.image.filename)"/>
```
Here you would replace `img.src.path` with `logo.src.path`.

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,259 @@
+                            <filename>graphic-pageheader/graphic-pageheader.xsl</filename>
+                        </output>
+                    </info>
+                    <programlisting language="xml">&lt;?xml version="1.0" encoding="UTF-8"?>
```
This is dangerous: you have on one side the file `graphic-pageheader/graphic-pageheader.xsl` and the content inside `programlisting`. This is risky as it leads to inconsistencies in both files.

I would suggest to use `xi:include` inside programlisting, you forgot that what I described in comment https://github.com/tomschr/dbcookbook/pull/44#discussion_r199356664 :wink: I've corrected the code with your current stylesheet name.

TODO: you need to add correct callouts.

```xml
<example xml:id="ex.fo.XXXXX">
   <title>Some good title (<filename>graphic-pageheader.xsl</filename>)</title>
   <info>
        <output xmlns="urn:x-toms:docbook-ext">
          <filename>graphic-pageheader/graphic-pageheader.xsl</filename>
        </output>
   </info>
   <programlistingco>
      <areaspec>
            <!-- Add your callouts through <area/> -->
      </areaspec>
      <programlisting language="xml" linenumbering="numbered"
          ><xi:include href="graphic-pageheader/graphic-pageheader.xsl" parse="text"
          /></programlisting>
   </programlistingco>
</example>
```

tomschr replied:
I've experimented with the `areaspec` element. You can use this snippet:

```xml
<areaspec>
     <area xml:id="co.topic.graphic-pageheader.path" coords="8"/>
     <area xml:id="co.topic.graphic-pageheader.content" coords="13"/>
     <area xml:id="co.topic.graphic-pageheader.variables" coords="14"/>
     <area xml:id="co.topic.graphic-pageheader.block" coords="18"/>
     <area xml:id="co.topic.graphic-pageheader.blank" coords="23"/>
     <area xml:id="co.topic.graphic-pageheader.center" coords="35"/>
     <area xml:id="co.topic.graphic-pageheader.right" coords="39"/>
     <area xml:id="co.topic.graphic-pageheader.height" coords="41 64" units="linecolumn"/>
     <area xml:id="co.topic.graphic-pageheader.attribute" coords="42"/>
     <area xml:id="co.topic.graphic-pageheader.template" coords="43 75" units="linecolumn"/>
     <area xml:id="co.topic.graphic-pageheader.file" coords="44 122" units="linecolumn"/>
</areaspec>
```

der-schmelle2 replied:
Done.

der-schmelle2 replied:
I've added your recommendation, but I'm not sure if it's correct. :-)

der-schmelle2 replied:
I modified the line numbers a little bit because I've added a comment line.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,259 @@
+    </section>
+    <section role="discussion">
+        <title>Discussion</title>
+        <para>These points will be submitted later</para>
```
Here you could describe:

* Put logo(s) inside a different directory and assign it with your new parameter `logo.src.path`.

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,259 @@
+                    /></para>
+            </listitem>
+            <listitem>
+                <para>Parameter description for <tag>xsl:attribute</tag><link
```
I'm not sure why you've add this, but IMHO this isn't really needed.

der-schmelle2 replied:
Removed.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,259 @@
+                    /></para>
+            </listitem>
+            <listitem>
+                <para>Parameter description for <tag>xsl:with-param</tag>
```
Same here, I would also remove it.

der-schmelle2 replied:
Removed.

der-schmelle2 committed:
Clarify the global option 'img.src.path'

tomschr reviewed (commented):
Thanks Peter for the contribution. I have some further things that you could implement. Nothing difficult.

Regarding the Discussion, I would do that later.

der-schmelle2 committed:
Conducted the proposed changes by @tomschr

See changed lines for more information

tomschr commented on en/xml/fo/graphic-pageheader/graphic-pageheader.xsl:
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0"
+  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+  xmlns:d="http://docbook.org/ns/docbook">
```
You are missing the fo namespace. The root element has to be:

```xsl
<xsl:stylesheet version="1.0"
  xmlns:fo="http://www.w3.org/1999/XSL/Format"
  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
  xmlns:d="http://docbook.org/ns/docbook">
```

der-schmelle2 replied:
Funny! I've taken the proposed header from your topic.permalinks.xml.... ;-)
I'll correct this and the previous recommendations.

tomschr replied:
Yes, but the permalinks topic is for HTML. :wink:

der-schmelle2 replied:
Done.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,266 @@
+                                    <tag>imagedata</tag> command does not need a
+                                path if all files are in the same folder. You
+                                can write something like this:</para>
+                            <para>&lt;imagedata fileref="pic02.pdf"/>
```
Use `screen` instead of `para`.

der-schmelle2 replied:
No longer appreciale, because we are now using logo.src.path instead of img.src.path.

der-schmelle2 committed:
Conducted the proposed changes II

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,231 @@
+                        linkend="ex.fo.graphic-pageheader"/> into your
+                    customization layer:</para>
+                    <example xml:id="ex.fo.graphic-pageheader">
+                        <title>Placing a logo in the right page header (<filename>graphic-pageheader.xsl</filename>)</title>
```
Use title style here: "Placing a Logo in the Right Page Header"

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,231 @@
+                    </example>
+                    <calloutlist>
+                        <callout arearefs="co.topic.graphic-pageheader.path">
+                            <para>This setting is unique for your logo.</para>
```
I would rephrase it as: "The directory path to your logos"

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,231 @@
+    </section>
+    <section role="discussion">
+        <title>Discussion</title>
+        <para>In line 8 of our style sheet (
```
I would be careful with phrases like "In line X...". Usually, this can be outdated very soon. Better refer to the callout bug (you can use `<xref linkend="..."/>`).

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,231 @@
+            be independent of the global image source path 
+             <tag>img.src.path</tag> parameter for pictures, graphics and so 
+            on. This new parameter is used in line 45 to build the path
+             for our logo.</para>
```
Actually, this is written more from an implementation point of view. If you change to the user's view, it could like this:

```
The stylesheet <filename>graphic-pageheader.xsl</filename> creates a logo in the right edge of your page
and uses the parameter <parameter>logo.src.path</parameter> to find the logo. By default, a logo is stored
in <filename>logos/<filename>. If your logos are in a different location, change it with your the XSLT processor.
For example, the following line uses ...
```

tomschr reviewed (commented):
It looks good, I have a small comment.

der-schmelle2 committed:
Discussion section rewritten

der-schmelle2 committed:
Fixed two typos
* Added the source of the solution

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -191,14 +180,19 @@
+                linkend="co.topic.graphic-pageheader.file"/> to find the right
+            logo.
+            <screen>&lt;xsl:param name="logo.src.path">logos/&lt;/xsl:param>
+            </screen></para>
```
Remove the linebreak between line 193 and 194.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,227 @@
+                    /></para>
+            </listitem>
+        </itemizedlist>
+        <para>__________</para>
```
I'm not sure what the purpose of this line.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,227 @@
+            <screen>&lt;xsl:param name="logo.src.path">logo/&lt;/xsl:param>
+            </screen></para>
+        <para/>
+        <para>These points will be submitted later</para>
```
I'm still confused by this line. Is it a comment for you? Then it should go into a `<remark>`. Is it for the readers? If you want to add additional references for other topics? In that case, refer to them with an `xref` in the "See Also" section.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,227 @@
+        <para>The <emphasis>Opensource</emphasis>-Logo is a trademark which is
+            owned by <link xlink:href="https://opensource.org"/>. This site is
+            not affiliated with or endorsed by the Open Source
+            Initiative.</para>
```
Hmn, thinking if this line is better covered with a legalnotice. For the time being, just let it here.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,227 @@
+                linkend="co.topic.graphic-pageheader.file"/> to find the right
+            logo.
+            <screen>&lt;xsl:param name="logo.src.path">logo/&lt;/xsl:param>
+            </screen></para>
```
* Remove the linebreak between line 195 and 196.
* Put your `screen` outside of the `para`.

tomschr reviewed (commented):
Found only some minor things. :wink:

der-schmelle2 committed:
Conducted the proposed changes III
* Removed the remark for further Topics.
* Line 206 is a distinction for the OSI Logo Trademark. This is a requirement if you use the OSI logo. If you would like to use another logo, let me know.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -189,24 +189,9 @@
-                <para>Logo and chapter title in the page header</para>
-            </listitem>
-        </itemizedlist>
+            This parameter is used within the position test procedure<xref linkend="co.topic.graphic-pageheader.file"/> to find the right
```
On minor detail: Remove the word "procedure" as it is already covered within the `xref`. Otherwise looks good!

tomschr commented:
@der-schmelle2 I've created a PDF of your topic, see [topic.graphic-pageheader.pdf](https://github.com/tomschr/dbcookbook/files/2196652/topic.graphic-pageheader.pdf). The layout of the PDF can be improved, but let's focus on the content. 

From what I've seen it looks good, but I would suggest the following changes:

* Use the element `filename` for `pagesetup.xsl` (first paragraph in the Solution section).
* I'm not sure if the the 2nd sentence in the 2nd paragraph is true.
* Step 2 and step 3 both start with the same sentence. Please adjust. :wink: 
* style sheet -> stylesheet

Hope that helps.

If you fix the above issues, I'll merge it into develop. :wink: Thanks! :+1:

der-schmelle2 committed:
Removed 'procedure' in line 192

der-schmelle2 committed:
oXygenXML line wrap a char 78

der-schmelle2 committed:
oXygenXML view changes for 4K display

tomschr commented on DocBook-Cookbook.xpr:
```diff
@@ -1,1688 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
@der-schmelle2 Did you delete this file by accident? :wink:

der-schmelle2 replied:
Hi Tom,
I'm sorry for that.
Today I'm working on a new Windows Laptop with oXygenXML and Gitkraken. Gitkraken told me that I had changed the Docbook-Cookbook.xpr (which was correct). I set this file to ignore and Gitkraken tied to delete it. I don't know why it was deleted from the repo.
Maybe it's a better idea to stash the file.

Peter

tomschr replied:
No problem, don't worry. Such things can happen. :smiley:  It's easy to fix.

If you want to ignore the file, you can use a "global Git ignore" file like this:

```
$ git config --global core.excludesfile '~/.gitignore'
$ echo DocBook-Cookbook.xpr >> ~/.gitignore
```

der-schmelle2 replied:
Hi Tom,
thanks for the PDF output.
As discussed today I'll rewrite the hole section under a new feature branch due to more knowledge I received today. 😊

der-schmelle2 committed:
Merge branch 'develop' of https://github.com/tomschr/dbcookbook.git

der-schmelle2 committed:
Merge remote-tracking branch 'origin/develop' into feature/pageheader-graphic

der-schmelle2 committed:
Cleanup

der-schmelle2 committed:
test

der-schmelle2 committed:
Rewritten with new header.table content

der-schmelle2 commented:
Hole section rewritten. Description of template header.table added.

tomschr commented on en/xml/fo/graphic-pageheader/db-graphic-pageheader.xsl:
```diff
@@ -0,0 +1,11 @@
+  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+  xmlns:d="http://docbook.org/ns/docbook">
+  
+  <xsl:import href="https://cdn.docbook.org/release/xsl/current/xhtml/docbook.xsl"/>
```
This URL looks strange. Shouldn't it be ".../**fo**/docbook.xsl"?

tomschr commented on DocBook-Cookbook.xpr:
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project version="20.0">
+<project version="20.1">
```
This file shouldn't be modified in this PR. Unfortunately, oXygen does it automatically. Can you revert the change of this file?

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+            inch high to avoid conflicts with the body text area. Furthermore,
+            we will place the document title on the opposite of the logo,
+            centered to the logo high. The template for our solution is obtained
+            from 'pagesetup.xsl' style sheet.</para>
```
* Use the element `filename` around `pagesetup.xsl`.
* style sheet -> stylesheet

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+            or a simple forwarding of parts of your document.</para>
+        <para>First of all you need to create a customization layer as described
+            in the following steps. If you have a customization layer style
+            sheet, you can skip the first step, otherwise follow the steps to
```
* style sheet -> stylesheet

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+                        linkend="dbc.common.dbcustomize"/>.</para>
+            </step>
+            <step>
+                <para>Include the style sheet from <xref
```
style sheet -> stylesheet

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+                    logo and text placing.</para>
+                <para>If you don't want to add only the first part of our
+                    solution and later the second one, you can include the hole
+                    style sheet in the next step.</para>
```
I think, this step isn't a "real" step. A real step focus on the action(s) that the user has to perform. IMHO, the above three paragraphs are explanations which should be moved **before** the procedure

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+                    style sheet in the next step.</para>
+            </step>
+            <step>
+                <para>Include the style sheet <xref
```
style sheet -> stylesheet

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+                <para>Include the style sheet <xref
+                        linkend="ex.fo.graphic-pageheader"/> into your
+                    customization layer and modify it to fit your needs.</para>
+                <para>Subsequent, we will explain the options we will focus on:</para>
```
I think, this paragraph is unnecessary.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+                    </callout>
+                    <callout arearefs="co.topic.graphic-pageheader.center">
+                        <para>In this demo, the center position has no content,
+                            but you can add a page number for example.</para>
```
I would rephase the above paragraph slightly to mention the element `<fo:page-number/>`:

```xml
... but you can add the page number with the element
<tag class="emptytag">fo:page-number</tag>, for example.
```

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+        <para>The stylesheet <filename>graphic-pageheader.xsl</filename> creates
+            a logo in the right edge of your page and uses the parameter
+                <parameter>logo.src.path</parameter> to find the logo. By
+            default, a logo is stored in <filename>logo/</filename>. If your
```
* a logo -> **the** logo
* in the ... directory.

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+            a logo in the right edge of your page and uses the parameter
+                <parameter>logo.src.path</parameter> to find the logo. By
+            default, a logo is stored in <filename>logo/</filename>. If your
+            logos are in a different location, change it with your XSLT
```
Can there be plural (logo**s**)?  Isn't only one logo used/placed in the header/footer?

tomschr reviewed (changes requested):
Thank you, great work! Some comments about the file, mostly typos and some questions.

tomschr commented:
@der-schmelle2: Here is the PDF of this topic. The layout is not pretty and the callouts are still missing. However, it should give you a better overview.

[Inserting Text and Logos into Page Headers and Footers](https://github.com/tomschr/dbcookbook/files/2421196/DBCookBook-v0.8.0-36-g56f952d.pdf)

tomschr commented on en/xml/fo/topic.graphic-pageheader.xml:
```diff
@@ -0,0 +1,232 @@
+            <listitem xml:id="li.fo.external-graphics">
+                <para>Full parameter description for
+                        <tag>fo:external-graphic</tag><link
+                        xlink:href="http://www.w3.org/TR/xsl/#fo_external-graphic"
```
Correct the `tag` element and remove `link`:

```xml
<tag xlink:href="http://www.w3.org/TR/xsl/#fo_external-graphic">fo:external-graphic</tag>
```

der-schmelle2 committed:
<tag/> correction

der-schmelle2 committed:
Project File changed

der-schmelle2 committed:
Intended shortend in file
